use-cases/using-Intel-QAT-and-kata.md

```diff
@@ -0,0 +1,484 @@
+```
+
+> **Note: There may be additional kernel parameters that need to be set to yes 
+> when compiling.**
+
+### Copy Kata Kernel
+
+```sh
+$ mkdir -p $KATA_KERNEL_LOCATION
+$ cp $LINUX_VER/arch/x86/boot/bzImage $KATA_KERNEL_LOCATION/vmlinuz-${LINUX_VER}_qat
+```
+
+### Prepare Kata root filesystem
+
+These instructions build upon the OS builder instructions located in the 
+[Developer Guide](../Developer-Guide.md). The following instructions use Clear
+Linux (Kata Containers default) as the root filesystem with systemd as the 
+init and will add in the `kmod` binary which is not a standard binary in a 
+Kata rootfs image. The kmod binary is necessary to load the QAT kernel modules
+when the virtual machine rootfs is booted. Docker should already be installed
+on your system before running the following commands. If a custom kata-agent
+needs to be used then please refer to the link above on how to add it in.
+
+```sh
+$ mkdir -p $OSBUILDER
+$ cd $OSBUILDER
+$ git clone https://github.com/kata-containers/osbuilder.git
+$ export ROOTFS_DIR=${OSBUILDER}/osbuilder/rootfs-builder/rootfs
+$ export EXTRA_PKGS='kmod'
+
+# Make Sure that the kata agent version matches the installed kata-runtime version
+$ export AGENT_VERSION=$(kata-runtime version | head -n 1 | grep -o "[0-9.]\+")
+$ cd ${OSBUILDER}/osbuilder/rootfs-builder
+$ sudo rm -rf ${ROOTFS_DIR}
+$ script -fec 'sudo -E GOPATH=$GOPATH USE_DOCKER=true SECCOMP=no ./rootfs.sh clearlinux'
+```
+
+### Compile Intel QAT drivers for Kata Containers Kernel and add to Kata Containers rootfs
+
+After the Kata Containers kernel is built with the proper configuration flags 
+the Intel QAT drivers must be built against that Kata Containers kernel 
+version in much the same way they were built above for the host OS. You must 
+set the `KERNEL_SOURCE_ROOT` variable to the Kata Containers kernel source 
+directory and build the Intel QAT drivers again.
+
+```sh
+$ cd $GOPATH
+$ export LINUX_VER=$(ls -d kata*)
+$ export KERNEL_MAJOR_VERSION=$(awk '/^VERSION =/{print $NF}' $GOPATH/$LINUX_VER/Makefile)
+$ export KERNEL_PATHLEVEL=$(awk '/^PATCHLEVEL =/{print $NF}' $GOPATH/$LINUX_VER/Makefile)
+$ export KERNEL_SUBLEVEL=$(awk '/^SUBLEVEL =/{print $NF}' $GOPATH/$LINUX_VER/Makefile)
+$ export KERNEL_EXTRAVERSION=$(awk '/^EXTRAVERSION =/{print $NF}' $GOPATH/$LINUX_VER/Makefile)
+$ export KERNEL_ROOTFS_DIR=${KERNEL_MAJOR_VERSION}.${KERNEL_PATHLEVEL}.${KERNEL_SUBLEVEL}${KERNEL_EXTRAVERSION}
+
+$ cd $QAT_SRC
+$ sudo -E make clean
+
+$ KERNEL_SOURCE_ROOT=$GOPATH/$LINUX_VER ./configure --disable-qat-lkcf --enable-icp-sriov=guest
+$ sudo -E make quickassist-all -j$(nproc)
+$ sudo -E make INSTALL_MOD_PATH=$ROOTFS_DIR qat-driver-install -j$(nproc)
+
+# usdm_drv also needs to be copied into rootfs modules path and depmod run
+$ sudo cp $QAT_SRC/build/usdm_drv.ko $ROOTFS_DIR/usr/lib/modules/${KERNEL_ROOTFS_DIR}/updates/drivers  
+
+# Note: usdm_drv will be loaded by configuring kata kernel boot parameter in kata config
```

Comment:
I'm personally not super-keen on comments in codeblocks since they cannot be spell-checked currently. However, please could you change this to "Kata".

Reply:
I can move them all out.  I'll do that and see how it looks. Originally I had these as snippets I could copy/paste and thought that was useful but prepending all my commands with $ means they have to be run individually.